tests/test_eigen.cpp

```diff
@@ -51,16 +118,46 @@ test_initializer eigen([](py::module &m) {
     mat << 0, 3, 0, 0, 0, 11, 22, 0, 0, 0, 17, 11, 7, 5, 0, 1, 0, 11, 0,
         0, 0, 0, 0, 11, 0, 0, 14, 0, 8, 11;
 
-    m.def("double_col", &double_col);
-    m.def("double_row", &double_row);
-    m.def("double_mat_cm", &double_mat_cm);
-    m.def("double_mat_rm", &double_mat_rm);
+    m.def("double_col", [](const Eigen::VectorXf &x) -> Eigen::VectorXf { return 2.0f * x; });
+    m.def("double_row", [](const Eigen::RowVectorXf &x) -> Eigen::RowVectorXf { return 2.0f * x; });
+    m.def("double_threec", [](py::EigenDRef<Eigen::Vector3f> x) { x *= 2; });
+    m.def("double_threer", [](py::EigenDRef<Eigen::RowVector3f> x) { x *= 2; });
+    m.def("double_mat_cm", [](Eigen::MatrixXf x) -> Eigen::MatrixXf { return 2.0f * x; });
+    m.def("double_mat_rm", [](DenseMatrixR x) -> DenseMatrixR { return 2.0f * x; });
     m.def("cholesky1", &cholesky1);
     m.def("cholesky2", &cholesky2);
     m.def("cholesky3", &cholesky3);
     m.def("cholesky4", &cholesky4);
-    m.def("cholesky5", &cholesky5);
-    m.def("cholesky6", &cholesky6);
+
+    // Mutators (Eigen maps into numpy variables):
+    m.def("add_rm", add_rm); // Only takes row-contiguous
+    m.def("add_cm", add_cm); // Only takes column-contiguous
+    m.def("add_any", add_any); // Accepts anything
+    // Overloaded versions that will accept either row or column contiguous:
+    m.def("add1", add_rm);
+    m.def("add1", add_cm);
+    m.def("add2", add_cm);
+    m.def("add2", add_rm);
+
+    // Return mutable references (numpy maps into eigen varibles)
+    m.def("get_cm_ref", []() { return Eigen::Ref<Eigen::MatrixXd>(get_cm()); });
+    m.def("get_rm_ref", []() { return Eigen::Ref<MatrixXdR>(get_rm()); });
+    // The same references, but non-mutable (numpy maps into eigen variables, but is !writeable)
+    m.def("get_cm_const_ref", []() { return Eigen::Ref<const Eigen::MatrixXd>(get_cm()); });
+    m.def("get_rm_const_ref", []() { return Eigen::Ref<const MatrixXdR>(get_rm()); });
+    // Just the corners (via a Map instead of a Ref):
+    m.def("get_cm_corners", get_cm_corners);
+    m.def("get_cm_corners_const", get_cm_corners_const);
```

Comment:
I was thinking everywhere: `choleskyn`, `incr_matrix`, `even_rows` etc. As far as I can see only `add_rm()` and `add_cm()` are ever used more than once.